### DISCUSSION TITLE
Is a public ip needed to build p2p network using pion webrtc?

### DISCUSSION THREAD

**Discussion by suzanneckerr** [pion/webrtc#1769] [Q&A] [answered]
I am trying to use webrtc to build a p2p chat app but I am bit confused. Do I still need to establish a node with a public ip to bootstrap the network?

**Comment by at-wat** [comment, marked as answer]
Usually, public IP is not needed as ICE automatically establishes connection over NATs.
However, some types of the NATs (including Symmetric NATs) makes it hard.

You can discover the type of the NAT by https://github.com/enobufs/go-nats.

**Comment by Sean-Der** [comment]
To learn a little bit more about the issues that come up with P2P networking also check out the [Connecting](https://webrtcforthecurious.com/docs/03-connecting/) chapter in WebRTC for the Curious.

If you want to do P2P you will probably need to run a [TURN Server](https://github.com/pion/turn). If you are just connecting browsers you probably don't even need Pion WebRTC.

As you scale/hit challenge with P2P check out the [other topologies](https://webrtcforthecurious.com/docs/08-applied-webrtc/#webrtc-topologies) that you can do with WebRTC.

**Reply by suzanneckerr** [reply]
Thank you for the comments provided.

**Comment by sallamy2580** [comment]
Recently I have experienced the similar problem so I understood what you want.
Please read this article. [here](https://androidforums.com/threads/how-to-get-peer-ip-address-on-p2p-chat.1194654/)
If you meant this is not a problem, please reach out to me so we can discuss further more.
